new/ohai-cookbook-segment.md

```diff
@@ -0,0 +1,45 @@
+---
+RFC: unassigned
+Author: Lamont Granquist <lamont@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Title
+
+Support ohai plugins under an `ohai` top level directory in cookbooks.  Load all
+ohai plugins in all synchronized cookbooks after cookbook synchronization.
+
+## Motivation
+
+    As a Chef User,
+    I want to have my custom Ohai plugins loaded on first bootstrap,
+    So I don't have to run chef twice.
+
+    As a Chef User,
+    I want my Ohai plugins loaded before attributes and compile/converge mode,
+    So I can use them without worrying about cookbook execution ordering.
+
+    As a Chef User,
+    I want my Ohai plugins synchronized with my cookbooks,
+    So that I don't incure more unavoidable round-trips to the Chef Server.
+
+    As a Chef Developer,
+    I want Ohai plugins as a first-class object,
+    So that I don't have to compile recipes to discover templates that drop plugins.
+
+## Specification
+
+The "segments" of a cookbook will be extended to include an "ohai" segment.  In this
+segment there will be plugins which are intended to be copied to the Ohai `plugin_path`.  All files in this segment will be copied, recursively, maintaining directory
+structure.
```

Comment:
What's the protocol for resolving duplicate plugin file names? E.g., I have my own `hostname.rb` plugin file (hehe) under `cookbook/ohai/hostname.rb`. Would this conflict with or just override `lib/ohai/plugins/hostname.rb`? In the case of conflict, should we take the approach of the `ohai` cookbook and put custom plugins into their own directory?